pkg/ccl/backupccl/file_sst_sink.go

```diff
@@ -331,6 +331,66 @@ func (s *fileSSTSink) write(ctx context.Context, resp exportedSpan) (roachpb.Key
 	return resp.resumeKey, err
 }
 
+func (s *fileSSTSink) writeKey(
+	ctx context.Context,
+	prefix []byte,
+	key storage.MVCCKey,
+	value storage.MVCCValue,
+	start hlc.Timestamp,
+	end hlc.Timestamp,
+) error {
+	// Initialize the writer if needed.
+	if s.out == nil {
+		if err := s.open(ctx); err != nil {
+			return err
+		}
+	}
+	prefixedKey := key
+	prefixedKey.Key = append(prefix, prefixedKey.Key...)
+	// In the case of compaction, we only keep one flushed file in memory since we are not dealing with
+	// export responses from KV.
+	if len(s.flushedFiles) == 0 {
+		f := backuppb.BackupManifest_File{
+			Span: roachpb.Span{
+				Key:    prefixedKey.Key,
+				EndKey: prefixedKey.Key,
```

Comment:
So this part isn't quite right. If I write using the key, then the `BackupManifest_File` will contain the full key instead of just the table. So when I run `SHOW BACKUP FILES ...`, instead of the `BackupManifest_File` having a start/end of `[Tenant/3/Table/104/1, Tenant/3/Table/104/2)`, it will instead have `[Tenant/3/Table/104/1/<key>, Tenant/3/Table/104/1/<key>]`.

Since we aren't making KV export requests, not sure what the best approach for this would be though.

Reply:
just update the manfiest file end key every time the user calls `filesink.writekey`?

Reply:
Right, that's what is currently being done, it's just that because you are setting the bounds to be the keys themselves, the `BackupManifest_File` has key-specific bounds instead of index-specific ones. To be fair, I'm not entirely sure this even is an issue, and I'm sure key-specific bounds show up in normal backups whenever we are forced to flush mid-table, but just wanted to point it out in case it is.